src/Server/Repositories/ClaraJobsApi.cs

```diff
@@ -96,6 +97,27 @@ await Policy.Handle<Exception>()
                 }).ConfigureAwait(false);
         }
 
+        public async Task AddMetadata(Job job, Dictionary<string,string> metadata)
+        {
+            await Policy.Handle<Exception>()
+                .WaitAndRetryAsync(
+                    3,
+                    retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
+                    (exception, retryCount, context) =>
+                    {
+                        _logger.Log(LogLevel.Error, "Exception while adding job metadata: {exception}", exception);
+                    })
+                .ExecuteAsync(async () =>
+                {
+                    if (!JobId.TryParse(job.JobId, out JobId jobId))
+                    {
+                        throw new ApplicationException($"Invalid JobId provided: {job.JobId}");
```

Comment:
should this be an `ApplicationException` or an `ArgumentException`?

Reply:
Will fix in upcoming PR